### ISSUE TITLE
IP queue stuck (AIL operated behind proxy)

### ISSUE THREAD

**Issue by certrik** [ail-project/ail-framework#165]
Hi,
since the beginning of ingesting pastesites via pystemon the IP queue is stuck. It seems it not once was able to process the IP queue.
Can this be a problem because AIL is operated behind a proxy?
![grafik](https://github.com/ail-project/ail-framework/assets/42176839/5dc18363-291f-4544-8f3f-89f414818604)

**Comment by certrik** [comment]
I don't know if the error in modules.log somwhow corresponds with this problem.
```
2023-06-19 12:01:02,416 werkzeug ERROR:Error on request:
Traceback (most recent call last):
  File "/data/AIL/ail-framework/AILENV/lib/python3.9/site-packages/werkzeug/serving.py", line 364, in run_wsgi
    execute(self.server.app)
  File "/data/AIL/ail-framework/AILENV/lib/python3.9/site-packages/werkzeug/serving.py", line 327, in execute
    for data in application_iter:
  File "/data/AIL/ail-framework/AILENV/lib/python3.9/site-packages/werkzeug/wsgi.py", line 289, in __next__
    return self._next()
  File "/data/AIL/ail-framework/AILENV/lib/python3.9/site-packages/werkzeug/wrappers/response.py", line 32, in _iter_encoded
    for item in iterable:
  File "/data/AIL/ail-framework/var/www/modules/dashboard/Flask_dashboard.py", line 45, in event_stream
    for msg in pubsub.listen():
  File "/data/AIL/ail-framework/AILENV/lib/python3.9/site-packages/redis/client.py", line 1664, in listen
    response = self.handle_message(self.parse_response(block=True))
  File "/data/AIL/ail-framework/AILENV/lib/python3.9/site-packages/redis/client.py", line 1542, in parse_response
    response = self._execute(conn, try_read)
  File "/data/AIL/ail-framework/AILENV/lib/python3.9/site-packages/redis/client.py", line 1518, in _execute
    return conn.retry.call_with_retry(
  File "/data/AIL/ail-framework/AILENV/lib/python3.9/site-packages/redis/retry.py", line 49, in call_with_retry
    fail(error)
  File "/data/AIL/ail-framework/AILENV/lib/python3.9/site-packages/redis/client.py", line 1520, in <lambda>
    lambda error: self._disconnect_raise_connect(conn, error),
  File "/data/AIL/ail-framework/AILENV/lib/python3.9/site-packages/redis/client.py", line 1507, in _disconnect_raise_connect
    raise error
  File "/data/AIL/ail-framework/AILENV/lib/python3.9/site-packages/redis/retry.py", line 46, in call_with_retry
    return do()
  File "/data/AIL/ail-framework/AILENV/lib/python3.9/site-packages/redis/client.py", line 1519, in <lambda>
    lambda: command(*args, **kwargs),
  File "/data/AIL/ail-framework/AILENV/lib/python3.9/site-packages/redis/client.py", line 1540, in try_read
    return conn.read_response(disconnect_on_error=False)
  File "/data/AIL/ail-framework/AILENV/lib/python3.9/site-packages/redis/connection.py", line 874, in read_response
    response = self._parser.read_response(disable_decoding=disable_decoding)
  File "/data/AIL/ail-framework/AILENV/lib/python3.9/site-packages/redis/connection.py", line 347, in read_response
    result = self._read_response(disable_decoding=disable_decoding)
  File "/data/AIL/ail-framework/AILENV/lib/python3.9/site-packages/redis/connection.py", line 357, in _read_response
    raw = self._buffer.readline()
  File "/data/AIL/ail-framework/AILENV/lib/python3.9/site-packages/redis/connection.py", line 260, in readline
    self._read_from_socket()
  File "/data/AIL/ail-framework/AILENV/lib/python3.9/site-packages/redis/connection.py", line 213, in _read_from_socket
    raise ConnectionError(SERVER_CLOSED_CONNECTION_ERROR)
redis.exceptions.ConnectionError: Connection closed by server.
2023-06-19 12:01:30,315 root WARNING:AIL 063b3ec2-73ae-41ea-adcb-0b57b4ab4c3b started
```

**Comment by certrik** [comment]
Is the queue maybe always stuck if one doesn't provide IP-ranges in `/configs/core.cfg` at 
```
[IP]
# list of comma-separated CIDR that you wish to be alerted for. e.g:
#networks = 192.168.34.0/24,10.0.0.0/8,192.168.33.0/24
networks =
```
?

**Comment by Terrtia** [comment]
Hi @certrik !
This module need to be rewritten (It doesn't support IPv6). I might transform it to a new IP addresses tracker.
I pushed a fix  c02b9f2a815e9ca42a68cfe9b54e8d2519e235dd to pop the queue if the config is empty.
